### PR TITLE
Update lvgl ui and add support for MKS Robin E3P motherboard

### PR DESCRIPTION
**Pull request by makerbase-mks**
#### 1. Update mks_ui code
- More parameter can be setting by TS35 online, include:

     1) BabyStep setting:
           Print interface: Option->Babystep
           Need enable: #define BABYSTEPPING, #define BABYSTEP_ZPROBE_OFFSET in Configuration_adv.h

     2) Z-Offset setting:
           Settings->Config->Leveling settings->Nozzle-to-probe offsets settings->Z offset

     3) FILAMENT_RUNOUT_SENSOR setting:
           Settings->Config->Adavance settings->Filament settings

     4) AUTO LEVELING COMMAND setting:
           Settings->Config->Leveling settings->AutoLeveling command setting

     5) MESH_BED_LEVELING setting:
           Settings->Config->Leveling settings->Manual leveling coordinate settings

-  Add support for encoder, it's can selection menu and enter menu
     Settings->Config->Adavance settings->Rotary encoder setting->Is the encoder function used?
#### 2. Add support for MKS Robin E3P motherboard
  - MKS Robin E3P motherboard and MKS TS35-E3 display screen can be installed directly on Ender3 machine
  - MKS Robin E3P hardware: https://github.com/makerbase-mks/MKS-Robin-E3P

### CODE REVIEW

File: Marlin/src/MarlinCore.cpp [hunk 1]
```diff
@@ -920,6 +924,14 @@ void setup() {
     SERIAL_ECHO_MSG("start");
   #endif
 
+  #if BOTH(HAS_TFT_LVGL_UI, USE_WIFI_FUNCTION)
+    #define SERIAL_PORT_2 -1
+    mks_esp_wifi_init();
+    WIFISERIAL.begin(WIFI_BAUDRATE);
+    serial_connect_timeout = millis() + 1000UL;
+    while (/*!WIFISERIAL && */PENDING(millis(), serial_connect_timeout)) { /*nada*/ }
```
Comment (thinkyhead) [inline Marlin/src/MarlinCore.cpp:931]: This is going to add a full second of dead time to the startup. Are you sure you don't want to examine `WIFISERIAL` for a non-zero value that could indicate a very quick connection success?

File: Marlin/src/lcd/extui/lib/mks_ui/draw_about.cpp [hunk 2]
```diff
@@ -73,42 +74,46 @@ void lv_draw_about(void) {
...
+  //lv_obj_set_style(fw_version, &tft_style_label_rel);
+  //lv_label_set_text(fw_version, SHORT_BUILD_VERSION);
+  //lv_obj_align(fw_version, NULL, LV_ALIGN_CENTER, 0, -60);
 
   fw_type = lv_label_create(scr, NULL);
   lv_obj_set_style(fw_type, &tft_style_label_rel);
   lv_label_set_text(fw_type,
     #if MB(MKS_ROBIN_PRO)
-      "Firmware: Robin_Pro35"
+      "Firmware: Marlin-bugfix2.0.x-MKS-2.1.0"
```
Comment (thisiskeithb) [inline Marlin/src/lcd/extui/lib/mks_ui/draw_about.cpp]: Won't this show `bugfix2.0.x-MKS-2.1.0` in *all* tagged releases now? Using something like `SHORT_BUILD_VERSION` would probably be ideal if you are wanting to list the Marlin version.

Comment (thisiskeithb) [reply Marlin/src/lcd/extui/lib/mks_ui/draw_about.cpp]: ... or was the intention of `fw_type` to list the motherboard? If that's the case, then using `MOTHERBOARD` would be a better choice. Your `#else` has `CUSTOM_MACHINE_NAME`, so it's not clear what you're trying to display here.

File: Marlin/src/MarlinCore.cpp (hunk 1 above)
Comment (rhapsodyv) [reply Marlin/src/MarlinCore.cpp:931]: I think we can talk about wifi serial later, as it is hardcoded disabled by now. And there's a lot of thinks to review about MKS wifi module....

File: Marlin/src/lcd/extui/lib/mks_ui/draw_about.cpp (hunk 2 above)
Comment (rhapsodyv) [reply Marlin/src/lcd/extui/lib/mks_ui/draw_about.cpp]: I agree. It need to be fixed. The about must reflect the board/machine/version and not a hardcoded value.

**Comment by rhapsodyv** [comment]
I tested the code and it's working fine in my Nano V2. Babystep is working fine too!!

After fixing what @thisiskeithb pointed out in the about screen, I don't see any reasons to this PR don't get merged. For me it's seems fully functional!

For wifi support, I will talk with MKS, to understand how their wifi module works, and how we can make the code working with a generic esp too. But we can fix it later, now wifi is fully disabled.

**Review by rhapsodyv — CHANGES_REQUESTED** [review]
@thinkyhead the changes in `Marlin/src/lcd/tft/` files would be better if done in the TFT Refactoring PR. Those files were moved for another folder ( `Marlin/src/lcd/tft_io/` ) and I'm sure it will break git in the merge.

**Pull request merged**